src/view/RoundViewFactory.java

```diff
@@ -12,11 +20,20 @@
 public class RoundViewFactory {
     private RoundViewFactory() {}
 
-    public static RoundView create(RoundViewModel roundViewModel, SubmitAnswerViewModel submitAnswerViewModel,
-                                   SubmitAnswerGameDataAccessInterface gameDataAccessObject) {
-        SubmitAnswerController submitAnswerController = createSubmitAnswerUseCase(submitAnswerViewModel, gameDataAccessObject);
+    public static RoundView create(ViewManagerModel viewManagerModel,
+                                   RoundViewModel roundViewModel,
+                                   SubmitAnswerViewModel submitAnswerViewModel,
+                                   GameOverViewModel gameOverViewModel,
+                                   SubmitAnswerGameDataAccessInterface submitAnswerGameDataAccessInterface
+    ) {
 
-        return new RoundView(roundViewModel, submitAnswerViewModel, submitAnswerController);
+        // submitAnswerGameDataAccessInterface and FinishRoundGameDataAccessInterface are the same so cast to reduce repeated parameters passed to function
+        FinishRoundGameDataAccessInterface finishRoundGameDataAccessInterface = (FinishRoundGameDataAccessInterface) submitAnswerGameDataAccessInterface;
```

Comment:
I'm wondering if it is more appropriate in terms of CA to pass multiple parameters (one for each of `SubmitAnswerGameDataAccessInterface` and `FinishRoundGameDataAccessInterface`) instead of casting the first to the second. The way I see it is that `RoundViewFactory` shouldn't need to have the external knowledge that `SubmitAnswerGameDataAccessInterface` is the same as `FinishRoundGameDataAccessInterface` because in general, this might not always be the case.

With that said, I do see how passing the same argument twice can seem odd. What are your thoughts on this?

Reply:
Ya I agree, I originally had it pass gameDataAccessObject twice as a parameter to RoundViewFactory in the Main file and it also felt very awkward. Both situations are going to feel awkward but in terms of CA I believe you are correct and we should just pass the game parameter twice. As a possible solution, we can look into creating a common interface that the gameDataAccessObject implements (instead of it implementing multiple interfaces from each use case). In this case, each person would update this common interface with their needed methods which would handle this case where we have the same interface for two use cases. Let me know your thoughts?